src/darker/import_sorting.py

```diff
@@ -1,29 +1,53 @@
 import logging
 from pathlib import Path
-from typing import List, cast
+from typing import Dict, List, Optional, Tuple, Union, cast
+
+from black import find_project_root
 
 try:
     from isort import SortImports
+    from isort.settings import (
+        _update_with_config_file,
+        default,
+        from_path,
+    )
 except ImportError:
     SortImports = None
+    _update_with_config_file = None
+    default = None
+    from_path = None
 
 logger = logging.getLogger(__name__)
 
+IsortSettings = Dict[str, Union[bool, int, List[str], str, Tuple[str], None]]
+
+
+def get_isort_settings(src: Optional[Path], config: Optional[str]) -> IsortSettings:
+    if src and config is None:
+        project_root = find_project_root((str(src),))
+        settings: IsortSettings = from_path(str(project_root))
+        return settings
+
+    computed_settings: IsortSettings = default.copy()
+    if config:
+        _update_with_config_file(config, ('tool.isort',), computed_settings)
+    return computed_settings
+
+
+def apply_isort(
+    content: str,
+    src: Optional[Path] = None,
+    config: Optional[str] = None,
+    line_length: Optional[int] = None,
+) -> str:
+    isort_settings = get_isort_settings(src, config)
+    if line_length:
+        isort_settings["line_length"] = line_length
 
-def apply_isort(content: str) -> str:
     logger.debug(
-        "SortImports(file_contents=..., check=True, multi_line_output=3, "
-        "include_trailing_comma=True, force_grid_wrap=0, use_parentheses=True, "
-        "line_length=88, quiet=True)"
-    )
-    result = SortImports(
-        file_contents=content,
-        check=True,
-        multi_line_output=3,
-        include_trailing_comma=True,
-        force_grid_wrap=0,
-        use_parentheses=True,
-        line_length=88,
-        quiet=True,
+        "SortImports(file_contents=..., check=True, {})".format(
+            ", ".join(f"{k}={v}" for k, v in isort_settings.items())
```

Comment:
This becomes seriously verbose in debug mode:
```
DEBUG:darker.import_sorting:SortImports(file_contents=..., check=True, force_to_top=[], skip=[], skip_glob=[],
line_length=88, wrap_length=0, line_ending=None, sections=('FUTURE', 'STDLIB', 'THIRDPARTY', 'FIRSTPARTY',
'LOCALFOLDER'), no_sections=False, known_future_library=['__future__'], known_standard_library=['AL', 'BaseHTTPServer',
'Bastion', 'CGIHTTPServer', 'Carbon', 'ColorPicker', 'ConfigParser', 'Cookie', 'DEVICE', 'DocXMLRPCServer',
'EasyDialogs', 'FL', 'FrameWork', 'GL', 'HTMLParser', 'MacOS', 'MimeWriter', 'MiniAEFrame', 'Nav', 'PixMapWrapper',
'Queue', 'SUNAUDIODEV', 'ScrolledText', 'SimpleHTTPServer', 'SimpleXMLRPCServer', 'SocketServer', 'StringIO', 'Tix',
'Tkinter', 'UserDict', 'UserList', 'UserString', 'W', '__builtin__', 'abc', 'aepack', 'aetools', 'aetypes', 'aifc',
'al', 'anydbm', 'applesingle', 'argparse', 'array', 'ast', 'asynchat', 'asyncio', 'asyncore', 'atexit', 'audioop',
'autoGIL', 'base64', 'bdb', 'binascii', 'binhex', 'bisect', 'bsddb', 'buildtools', 'builtins', 'bz2', 'cPickle',
'cProfile', 'cStringIO', 'calendar', 'cd', 'cfmfile', 'cgi', 'cgitb', 'chunk', 'cmath', 'cmd', 'code', 'codecs',
'codeop', 'collections', 'colorsys', 'commands', 'compileall', 'compiler', 'concurrent', 'configparser', 'contextlib',
'contextvars', 'cookielib', 'copy', 'copy_reg', 'copyreg', 'crypt', 'csv', 'ctypes', 'curses', 'dataclasses',
'datetime', 'dbhash', 'dbm', 'decimal', 'difflib', 'dircache', 'dis', 'distutils', 'dl', 'doctest', 'dumbdbm',
'dummy_thread', 'dummy_threading', 'email', 'encodings', 'ensurepip', 'enum', 'errno', 'exceptions', 'faulthandler',
'fcntl', 'filecmp', 'fileinput', 'findertools', 'fl', 'flp', 'fm', 'fnmatch', 'formatter', 'fpectl', 'fpformat',
'fractions', 'ftplib', 'functools', 'future_builtins', 'gc', 'gdbm', 'gensuitemodule', 'getopt', 'getpass', 'gettext',
'gl', 'glob', 'grp', 'gzip', 'hashlib', 'heapq', 'hmac', 'hotshot', 'html', 'htmlentitydefs', 'htmllib', 'http',
'httplib', 'ic', 'icopen', 'imageop', 'imaplib', 'imgfile', 'imghdr', 'imp', 'importlib', 'imputil', 'inspect', 'io',
'ipaddress', 'itertools', 'jpeg', 'json', 'keyword', 'lib2to3', 'linecache', 'locale', 'logging', 'lzma', 'macerrors',
'macostools', 'macpath', 'macresource', 'mailbox', 'mailcap', 'marshal', 'math', 'md5', 'mhlib', 'mimetools',
'mimetypes', 'mimify', 'mmap', 'modulefinder', 'msilib', 'msvcrt', 'multifile', 'multiprocessing', 'mutex', 'netrc',
'new', 'nis', 'nntplib', 'numbers', 'operator', 'optparse', 'os', 'ossaudiodev', 'parser', 'pathlib', 'pdb', 'pickle',
'pickletools', 'pipes', 'pkgutil', 'platform', 'plistlib', 'popen2', 'poplib', 'posix', 'posixfile', 'pprint',
'profile', 'pstats', 'pty', 'pwd', 'py_compile', 'pyclbr', 'pydoc', 'queue', 'quopri', 'random', 're', 'readline',
'reprlib', 'resource', 'rexec', 'rfc822', 'rlcompleter', 'robotparser', 'runpy', 'sched', 'secrets', 'select',
'selectors', 'sets', 'sgmllib', 'sha', 'shelve', 'shlex', 'shutil', 'signal', 'site', 'sitecustomize', 'smtpd',
'smtplib', 'sndhdr', 'socket', 'socketserver', 'spwd', 'sqlite3', 'ssl', 'stat', 'statistics', 'statvfs', 'string',
'stringprep', 'struct', 'subprocess', 'sunau', 'sunaudiodev', 'symbol', 'symtable', 'sys', 'sysconfig', 'syslog',
'tabnanny', 'tarfile', 'telnetlib', 'tempfile', 'termios', 'test', 'textwrap', 'this', 'thread', 'threading', 'time',
'timeit', 'tkinter', 'token', 'tokenize', 'trace', 'traceback', 'tracemalloc', 'ttk', 'tty', 'turtle', 'turtledemo',
'types', 'typing', 'unicodedata', 'unittest', 'urllib', 'urllib2', 'urlparse', 'usercustomize', 'uu', 'uuid', 'venv',
'videoreader', 'warnings', 'wave', 'weakref', 'webbrowser', 'whichdb', 'winreg', 'winsound', 'wsgiref', 'xdrlib', 'xml',
'xmlrpc', 'xmlrpclib', 'zipapp', 'zipfile', 'zipimport', 'zlib'], known_third_party=['google.appengine.api', 'py',
'pytest'], known_first_party=[], multi_line_output=3, forced_separate=[], indent= , comment_prefix= #,
length_sort=False, add_imports=[], remove_imports=[], reverse_relative=False, force_single_line=False,
default_section=FIRSTPARTY, import_heading_future=, import_heading_stdlib=, import_heading_thirdparty=,
import_heading_firstparty=, import_heading_localfolder=, balanced_wrapping=False, use_parentheses=True,
order_by_type=True, atomic=False, lines_after_imports=-1, lines_between_sections=1, lines_between_types=0,
combine_as_imports=False, combine_star=False, keep_direct_and_as_imports=False, include_trailing_comma=True,
from_first=False, verbose=False, quiet=False, force_adds=False, force_alphabetical_sort_within_sections=False,
force_alphabetical_sort=False, force_grid_wrap=0, force_sort_within_sections=False, show_diff=False,
ignore_whitespace=False, no_lines_before=[], no_inline_sort=False, ignore_comments=False, safety_excludes=True,
case_sensitive=False)
```

Also strings are unquoted, so at least this would be appropriate:
```suggestion
            ", ".join(f"{k}={v!r}" for k, v in isort_settings.items())
```

I experimented with an abbreviation mechanism as well, adding this function:
```python
    def abbreviate(value):
        if isinstance(value, (tuple, list)) and len(value) > 2:
            return "..."
        return repr(value)
```
and using it here:
```suggestion
            ", ".join(f"{k}={abbreviate(v)}" for k, v in isort_settings.items())
```

With that, debug output shrinks to:
```
DEBUG:darker.import_sorting:SortImports(file_contents=..., check=True, force_to_top=[], skip=[], skip_glob=[],
line_length=88, wrap_length=0, line_ending=None, sections=..., no_sections=False, known_future_library=['__future__'],
known_standard_library=..., known_third_party=..., known_first_party=[], multi_line_output=3, forced_separate=[],
indent=' ', comment_prefix=' #', length_sort=False, add_imports=[], remove_imports=[], reverse_relative=False,
force_single_line=False, default_section='FIRSTPARTY', import_heading_future='', import_heading_stdlib='',
import_heading_thirdparty='', import_heading_firstparty='', import_heading_localfolder='', balanced_wrapping=False,
use_parentheses=True, order_by_type=True, atomic=False, lines_after_imports=-1, lines_between_sections=1,
lines_between_types=0, combine_as_imports=False, combine_star=False, keep_direct_and_as_imports=False,
include_trailing_comma=True, from_first=False, verbose=False, quiet=False, force_adds=False,
force_alphabetical_sort_within_sections=False, force_alphabetical_sort=False, force_grid_wrap=0,
force_sort_within_sections=False, show_diff=False, ignore_whitespace=False, no_lines_before=[], no_inline_sort=False,
ignore_comments=False, safety_excludes=True, case_sensitive=False)
```

It's still quite a bit. Maybe we should replace all of those with a `, ...`, or if we want to go through the trouble, only include items which differ from the defaults. What's your opinion?